packages/go_router/test/src/go_router_cupertino_test.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+import 'package:go_router/src/go_router_cupertino.dart';
+
+void main() {
+  group('isCupertinoApp', () {
+    testWidgets('returns [true] when CupertinoApp is present', (tester) async {
+      final key = GlobalKey<_DummyStatefulWidgetState>();
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: DummyStatefulWidget(key: key),
+        ),
+      );
+      final isCupertino = isCupertinoApp(key.currentContext! as Element);
+      expect(isCupertino, true);
+    });
+
+    testWidgets('returns [false] when MaterialApp is present', (tester) async {
+      final key = GlobalKey<_DummyStatefulWidgetState>();
+      await tester.pumpWidget(
+        MaterialApp(
+          home: DummyStatefulWidget(key: key),
+        ),
+      );
+      final isCupertino = isCupertinoApp(key.currentContext! as Element);
+      expect(isCupertino, false);
+    });
+  });
+
+  test('pageBuilderForCupertinoApp creates a [CupertinoPage] accordingly', () {
+    final key = UniqueKey();
+    const name = 'name';
+    const arguments = 'arguments';
+    const restorationId = 'restorationId';
+    const child = DummyStatefulWidget();
+    final page = pageBuilderForCupertinoApp(
+      key: key,
+      name: name,
+      arguments: arguments,
+      restorationId: restorationId,
+      child: child,
+    );
+    expect(page.key, key);
```

Comment:
also assert page is cupertinoPage?

Reply:
I don't this is necessary because the return type of `pageBuilderForCupertinoApp` is already a `CupertinoPage<void>`, so we're safe.

Reply:
Since the custom lint rule, this will still pass if we change pageBuilderForCupertinoApp to return a different page class. The type need to be forced if go_router uses the flutter package lint

Reply:
so I will say lets wait for the lint pr to be merged before this one

Reply:
So with the lint rule it's not needed to change anything?